tempesta_fw/http.c

```diff
@@ -689,6 +701,14 @@ tfw_http_req_process(TfwConnection *conn, struct sk_buff *skb, unsigned int off)
 		tfw_cache_req_process((TfwHttpReq *)hmreq,
 				      tfw_http_req_cache_cb);
 
+		/*
+		 * At this stage the connection may have been dropped
+		 * and the socket closed. Further processing of data
+		 * in the SKB makes little sense.
+		 */
+		if (unlikely(!ss_sock_active(req_sk)))
+			return TFW_PASS;
```

Comment:
Pessimism is bad. It's better to optimistically go further and correctly handle connection drop: anyway connection can be closed in any given time.